app/assets/javascripts/sidebar.js

```diff
@@ -1,29 +1,124 @@
+    // Not available in all the layouts, so this selector might not be present. Again, avoid overlapping.
+    var $sideNavigation = $('#nav-activity-menu');
+    if ($sideNavigation.length > 0) {
+      minOffset = Math.max(minOffset, $sideNavigation[0].getBoundingClientRect().bottom)
+    }
+    minOffset = minOffset + SIDEBAR_SPACER; // add a little margin, it looks better.
+    // Then, make sure that multiple handles don't overlap and they don't go off screen.
+    var sidebarSpacing = $('.sidebar-hdr').height() + SIDEBAR_SPACER;
+    var titleBarHeight = $('.sidebar-mod .title-bar').height();
     $('.sidebar-mod').each(function (idx) {
-      $(this).css('top', sidebarTop + idx * (sidebarHdrHeight + SIDEBAR_SPACER));
+      const top = minOffset + idx * sidebarSpacing;
+      $(this).css('top', top);
+      // Also, ensure that sidebar content is fully visible, even on the pretty short screens.
+      $(this).find('.sidebar-content').css('max-height', window.innerHeight - top - titleBarHeight - BOTTOM_MARGIN);
     });
   }
 
-  if ($('.sidebar-mod').length > 0) {
-    positionMultipleSidebars();
+  window.Sidebar = {
+    // List of all existing sidebars (their controllers).
+    _controllers: [],
 
-    $('.sidebar-hdr').add('.sidebar-bd-close').click(function () {
-      $(this).closest('.sidebar-mod').toggleClass('expanded');
-    });
-    // It triggers CSS transition.
-    $('.sidebar-mod').addClass('visible');
-  }
-});
\ No newline at end of file
+    _closeAllSidebars: function () {
+      this._controllers.forEach(function (controller) {
+        controller.close();
+      });
+    },
+
+    // This function is meant to be used by LARA API. At this point there's no reason to use it directly.
+    // LARA API provides reasonable default options too.
+    addSidebar: function (options) {
+      // Generate HTML.
+      var $sidebar = $('<div class="sidebar-mod">');
+      var $handle = $('<div class="sidebar-hdr">');
+      var $body = $('<div class="sidebar-bd">');
+      // Handle.
+      if (options.icon) {
+        $handle.append(options.icon);
+      }
+      var $handleText = $('<h5 class="h5">');
+      $handle.append($handleText);
+      // Body / main container.
+      var $closeBtn = $('<button class="sidebar-bd-close">');
+      $closeBtn.button({ icon: "ui-icon-closethick" });
+      $body.append($closeBtn);
+      var $contentContainer = $('<div class="sidebar-content">');
+      $body.append($contentContainer);
+      // Final setup.
+      $sidebar.append($handle);
+      $sidebar.append($body);
+      $('body').append($sidebar);
+
+      // Add event handlers.
+      var isOpen = function () {
+        return $sidebar.hasClass('expanded')
+      };
+      $handle.add($closeBtn) // .add creates a set of elements, so we can apply click handler just once
+        .on('click.sidebarToggle', function () {
```

Comment:
I have a feeling I am about to learn something about jQuery here.  

What is this string 'click.sidebarToggle'?  It looks like a custom event, but I guess I am not sure how it works.

Is there something special about the string?  I can't seem to find other references in the source?

I couldn't find it in jQuery docs, but by playing with it in the console, it seems exactly the same as `on('click')`

Maybe we can have other elements listen for the special event `click.sidebarToggle`

Reply:
This is a namespace. It's quite similar in other even handling libraries (EventEmitter2 etc).

It's most useful when you want to cleanup event handlers. If you just do `$elem.off('click')`, it would remove all the click handlers, even ones added by other scripts. So, instead you can call `$elem.off('click.sidebarToggle')` to remove handlers that were added by you only. Or, `$elem.off('.sidebarToggle')` to remove all handlers with a given namespace (click, touch, mouseover, and so on).

I added it here, as this function used to be called multiple times for a single element and I had to call `.off` first. Since the implementation changed and I never use this namespace, yes, it's redundant now.